### PR TITLE
[clang-tidy] fix false positive in modernize-min-max-use-initializer-list

### LINKED ISSUE #107594: [clang-tidy] False positive modernize-min-max-use-initializer-list with mixed types
~~~c++
#include <algorithm>

int f(int a, int b, char c) {
    return std::max<int>({ a, b, c });
}
~~~
~~~
<source>:4:12: warning: do not use nested 'std::max' calls, use an initializer list instead [modernize-min-max-use-initializer-list]
    4 |     return std::max<int>({a, b, c});
      |            ^                    ~
      |                                 static_cast<int>(c)
~~~
https://godbolt.org/z/xnYx71nTe

### PR DESCRIPTION
**Pull request by 5chmidti**
Previously, whenever a replacement was generated by the analysis, a
diagnostic was generated. This became an issue when a call to
`std::min` or `std::max` consisted only of an initializer list with at
least one argument to the list requiring a type cast.
In this case, a single replacement that added a `static_cast` was created,
that resulted in a diagnostic being issued but with no nested call
to `std::min` or `std::max`.

Instead, explicitly track if a nested call was detected and only emit a
diagnostic if this is the case.

Fixes #107594

### CODE REVIEW

File: clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp [hunk 1]
```diff
@@ -168,9 +170,11 @@ generateReplacements(const MatchFinder::MatchResult &Match,
           CharSourceRange::getTokenRange(InnerResult.First->getEndLoc())));
     }
 
-    const SmallVector<FixItHint> InnerReplacements = generateReplacements(
+    const auto [FoundNestedCallInner, InnerReplacements] = generateReplacements(
         Match, InnerCall, InnerResult, IgnoreNonTrivialTypes,
-        IgnoreTrivialTypesOfSizeAbove);
+        IgnoreTrivialTypesOfSizeAbove, false);
+
+    FoundNestedCall |= FoundNestedCallInner;
```
Comment (SimplyDanny) [inline clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp:177]: Isn't the result always `true` with the value being reset to `true` in line 149 and having an `|=` here?

File: clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp [hunk 2]
```diff
@@ -72,11 +72,10 @@ static FindArgsResult findArgs(const CallExpr *Call) {
...
 
-static SmallVector<FixItHint>
-generateReplacements(const MatchFinder::MatchResult &Match,
-                     const CallExpr *TopCall, const FindArgsResult &Result,
-                     const bool IgnoreNonTrivialTypes,
-                     const std::uint64_t IgnoreTrivialTypesOfSizeAbove) {
+static std::pair<bool, SmallVector<FixItHint>> generateReplacements(
+    const MatchFinder::MatchResult &Match, const CallExpr *TopCall,
+    const FindArgsResult &Result, const bool IgnoreNonTrivialTypes,
+    const std::uint64_t IgnoreTrivialTypesOfSizeAbove, bool FoundNestedCall) {
```
Comment (SimplyDanny) [inline clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp:78]: All callers use `false` as the last argument. Is the new argument needed at all?

**Comment by 5chmidti** [comment]
You're right with both of your comments. I've changed the implementation in the latest commit.

**Review by SimplyDanny — APPROVED** [review]

File: clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ static FindArgsResult findArgs(const CallExpr *Call) {
   return Result;
 }
 
-static SmallVector<FixItHint>
+static std::pair<bool, SmallVector<FixItHint>>
```
Comment (HerrCai0907) [inline clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp:75]: Could you add some comments of this result. since bool means nothing.

**Review by HerrCai0907 — APPROVED** [review]
LGTM

File: clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp (hunk 3 above)
Comment (5chmidti) [reply clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp:79]: done

**Pull request merged**